pkg/auth/handlers/handlers.go

```diff
@@ -62,12 +65,29 @@ func NewRequestAuthenticator(mapper api.RequestContextMapper, auth authenticator
 			}
 
 			authenticatedUserCounter.WithLabelValues(compressUsername(user.GetName())).Inc()
+			setAuthenticationInfo(w, user)
 
 			handler.ServeHTTP(w, req)
 		}),
 	)
 }
 
+// set the "Authentication-Info" header for a response.
+func setAuthenticationInfo(w http.ResponseWriter, userInfo user.Info) {
+	// Requirements for quoting these values is defined in RFC 7235 (which links to RFC 7230).
+	// We use the strconv package because it encompasses the requirements, including escaping
+	// inner quotes, spaces, and non-printable characters.
+	//
+	// QuoteToASCII because non-ASCII HTTP headers aren't guarenteed to work with many clients.
+	//
+	// https://tools.ietf.org/html/rfc7235#section-2.1
+	// https://tools.ietf.org/html/rfc7230#section-3.2.6
+	username := strconv.QuoteToASCII(userInfo.GetName())
```

Comment:
I take it `QuoteToASCII` is different from `%q`?  If so, can you note that in your comment explicitly.